contracts/mocks/hts-precompile/HtsPrecompileMock.sol

```diff
@@ -368,185 +360,102 @@ contract HtsPrecompileMock is NoDelegateCall, IHederaTokenService, KeyHelper {
-            return HederaResponseCodes.INVALID_TOKEN_ID;
-        }
-
-        (bool validKey, bool noKey) = _hasAdminKeySig(token);
-
-        if (noKey) {
-            return HederaResponseCodes.TOKEN_IS_IMMUTABLE;
-        }
-
-        if (!validKey) {
-            return HederaResponseCodes.INVALID_ADMIN_KEY;
-        }
-
+    function _precheckUpdateTokenExpiryInfo(address sender, address token, Expiry memory expiryInfo) internal view returns (bool success, int64 responseCode) {
+        success = true;
+        (success, responseCode) = success ? Validation._validateToken(token, _tokenDeleted, _isFungible, _isNonFungible) : (success, responseCode);
+        (success, responseCode) = success ? _validateAdminKey(token) : (success, responseCode);
         // TODO: validate expiryInfo; move validation into common Validation contract that exposes validation functions
-
     }
 
-    function _precheckUpdateTokenInfo(address sender, address token, HederaToken memory tokenInfo) internal view returns (int64 responseCode) {
-
-        if (_tokenDeleted[token]) {
-            return HederaResponseCodes.TOKEN_WAS_DELETED;
-        }
-
-        CommonPrecheckData memory commonPrecheckData = _getCommonPrecheckData(token, sender, sender, ADDRESS_ZERO);
-
-        if (!commonPrecheckData.isFungible && !commonPrecheckData.isNonFungible) {
-            return HederaResponseCodes.INVALID_TOKEN_ID;
-        }
-
-        (bool validKey, bool noKey) = _hasAdminKeySig(token);
-
-        if (noKey) {
-            return HederaResponseCodes.TOKEN_IS_IMMUTABLE;
-        }
-
-        if (!validKey) {
-            return HederaResponseCodes.INVALID_ADMIN_KEY;
-        }
-
+    function _precheckUpdateTokenInfo(address sender, address token, HederaToken memory tokenInfo) internal view returns (bool success, int64 responseCode) {
+        success = true;
+        (success, responseCode) = success ? Validation._validateToken(token, _tokenDeleted, _isFungible, _isNonFungible) : (success, responseCode);
+        (success, responseCode) = success ? _validateAdminKey(token) : (success, responseCode);
         // TODO: validate tokenInfo; move validation into common Validation contract that exposes validation functions
-
     }
 
-    function _precheckUpdateTokenKeys(address sender, address token, TokenKey[] memory keys) internal view returns (int64 responseCode) {
+    function _precheckUpdateTokenKeys(address sender, address token, TokenKey[] memory keys) internal view returns (bool success, int64 responseCode) {
+        success = true;
+        (success, responseCode) = success ? Validation._validateToken(token, _tokenDeleted, _isFungible, _isNonFungible) : (success, responseCode);
+        (success, responseCode) = success ? _validateAdminKey(token) : (success, responseCode);
+        // TODO: validate keys; move validation into common Validation contract that exposes validation functions
```

Comment:
Another ticket potential and remove the comment once merged

Reply:
There are a lot more TODOs, which is what [task 1](https://github.com/hashgraph/hedera-json-rpc-relay/issues/1220) relates to:

> Conduct a project-wide search for all TODOs, documenting them as issues and prioritizing them accordingly.